rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactoryTest.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    private void setWidgetArgs() throws JSONException {
+        widgetArgs = new WidgetArgs();
+        RDTJsonFormFragment formFragment = mock(RDTJsonFormFragment.class);
+        widgetArgs.setFormFragment(formFragment);
+        widgetArgs.setStepName("step6");
+
+        String step = "{\n" +
+                "    \"title\": \"Sample collection\",\n" +
+                "    \"display_back_button\": \"true\",\n" +
+                "    \"next\": \"step8\",\n" +
+                "    \"bottom_navigation\": \"true\",\n" +
+                "    \"bottom_navigation_orientation\": \"vertical\",\n" +
+                "    \"fields\": [\n" +
+                "      {\n" +
+                "        \"key\": \"temp_sensor\",\n" +
+                "        \"openmrs_entity_parent\": \"\",\n" +
+                "        \"openmrs_entity\": \"\",\n" +
+                "        \"openmrs_entity_id\": \"\",\n" +
+                "        \"type\": \"hidden\",\n" +
+                "        \"value\": \"true\"\n" +
+                "      }\n" +
+                "    ]\n" +
+                "  }";
+
+        JSONObject jsonObject = new JSONObject(step);
+
+        widgetArgs.setJsonObject(jsonObject);
+
+        jsonFormActivity = mock(JsonFormActivity.class);
+        Resources resources = mock(Resources.class);
+        doReturn(resources).when(jsonFormActivity).getResources();
+        doReturn(0).when(resources).getColor(anyInt());
+        doReturn(jsonObject).when(jsonFormActivity).getStep(Mockito.eq("step6"));
+        widgetArgs.setContext(jsonFormActivity);
+        Whitebox.setInternalState(oneScanCovidRDTBarcodeFactory, "widgetArgs", widgetArgs);
+    }
+
+    private void mockStaticMethods() throws JSONException {
+
+        mockStatic(RDTApplication.class);
+        PowerMockito.when(RDTApplication.getInstance()).thenReturn(rdtApplication);
+        PowerMockito.when(rdtApplication.getStepStateConfiguration()).thenReturn(stepStateConfig);
+
+        JSONObject jsonObject = mock(JSONObject.class);
+        doReturn("step11").when(jsonObject).optString(eq(Constants.Step.SENSOR_TRIGGERED_PAGE), Mockito.eq("step1"));
+
+        doReturn(jsonObject).when(stepStateConfig).getStepStateObj();
+
+        Whitebox.setInternalState(oneScanCovidRDTBarcodeFactory, "stepStateConfig", stepStateConfig);
+    }
+
+    private void mockStaticClasses() {
+        mockStatic(LayoutInflater.class);
+        LayoutInflater layoutInflater = mock(LayoutInflater.class);
+        mockStatic(LayoutInflater.class);
+        RelativeLayout rootLayout = mock(RelativeLayout.class);
+        PowerMockito.when(LayoutInflater.from(any(Context.class))).thenReturn(layoutInflater);
+        doReturn(rootLayout).when(layoutInflater).inflate(anyInt(), isNull());
+        doReturn(mock(MaterialEditText.class)).when(rootLayout).findViewById(R.id.edit_text);
+        doReturn(mock(Button.class)).when(rootLayout).findViewById(R.id.scan_button);
+    }
```

Comment:
This mocking will not be necessary after https://github.com/onaio/rdt-standard/pull/463/commits/3be90ebdfadd9191c4b7cdcf2c1ba65db4ad7347#r485404727.

Reply:
fixed.